app/frontend/src/components/autocomplete/autocomplete.js

```diff
@@ -6,40 +7,47 @@ import api from '../../lib/api';
 
 const SHOW_SUGGESTIONS_THRESHOLD = 3;
 
-const highlightRefinement = (text, refinement) => {
-  const index = text.toLowerCase().indexOf(refinement.toLowerCase());
+const suggestionHTML = (text, inputValue) => {
+  const index = text.toLowerCase().indexOf(inputValue.toLowerCase());
 
   /* eslint-disable max-len */
-  return `${text.substring(0, index)}<span class='accessible-autocomplete__suggestion-highlight'>${text.substring(index, index + refinement.length)}</span>${text.substring(index + refinement.length, text.length)}`;
+  return `${text.substring(0, index)}<span class='accessible-autocomplete__suggestion-highlight'>${text.substring(index, index + inputValue.length)}</span>${text.substring(index + inputValue.length, text.length)}`;
   /* eslint-enable */
 };
 
-const autocomplete = () => {
-  Array.from(document.querySelectorAll('.autocomplete')).forEach((el) => {
-    const formInput = el.querySelector('input');
-    const dataSource = api[el.dataset.source];
+export default class extends Controller {
+  connect() {
+    const formInput = this.element.querySelector('input');
```

Comment:
Should we use a Stimulus' target API for this?

Reply:
i had a quandry about this tbh but i settled on no. thats because in this case its a stimulus controller being applied to a bit of html (input) that shouldnt/doesnt have knowledge of autocomplete being applied to it so it shouldnt have to explicitly define the input is a target.

Reply:
How come the input shouldn't have knowledge of autocomplete being applied to it?